bot/exts/help_channels/_channel.py

```diff
@@ -25,23 +25,58 @@ def get_category_channels(category: discord.CategoryChannel) -> t.Iterable[disco
             yield channel
 
 
-async def get_idle_time(channel: discord.TextChannel) -> t.Optional[int]:
-    """
-    Return the time elapsed, in seconds, since the last message sent in the `channel`.
-
-    Return None if the channel has no messages.
-    """
-    log.trace(f"Getting the idle time for #{channel} ({channel.id}).")
-
-    msg = await _message.get_last_message(channel)
-    if not msg:
-        log.debug(f"No idle time available; #{channel} ({channel.id}) has no messages.")
-        return None
-
-    idle_time = (datetime.utcnow() - msg.created_at).seconds
-
-    log.trace(f"#{channel} ({channel.id}) has been idle for {idle_time} seconds.")
-    return idle_time
+async def get_closing_time(channel: discord.TextChannel, init_done: bool) -> t.Tuple[datetime, str]:
+    """Return the timestamp at which the given help `channel` should be closed along with the reason."""
+    log.trace(f"Getting the closing time for #{channel} ({channel.id}).")
+
+    is_empty = await _message.is_empty(channel)
+
+    if is_empty:
+        idle_minutes = constants.HelpChannels.deleted_idle_minutes
+    else:
+        idle_minutes = constants.HelpChannels.idle_minutes_claimant
+
+    non_claimant_last_message_time = await _caches.non_claimant_last_message_times.get(channel.id)
+    claimant_last_message_time = await _caches.claimant_last_message_times.get(channel.id)
+
+    if is_empty or not all(
+        init_done,
+        non_claimant_last_message_time,
+        claimant_last_message_time,
+    ):
```

Comment:
Is this logic right? It should do it if either the cog is still initialising or the caches are empty.

Reply:
If any of the variables within the `all()` call are falsely, we want to enter the if block and use the fallback behaviour. IE if the cog is initialising or if either cache is empty

Reply:
Right... `¬(A & B & C) == (¬A | ¬B | ¬C)`